### PR TITLE
Some fixes for yandex music which led to an error when loading a track or playlist

### PR DESCRIPTION
**Pull request by BrokenFG**
Fix error when track doesn't have cover
Fix error when track is not available
Fix error when track doesn't include in any albums (for example user uploaded track manually)

### CODE REVIEW

File: main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java [hunk 1]
```diff
@@ -173,22 +173,29 @@ public String getDownloadStrings(String uri) throws IOException {
...
+            var parsedTrack = this.parseTrack(track);
+            if (parsedTrack != null) {
+                tracks.add(this.parseTrack(track));
+            }
         }
         return tracks;
     }
 
     private AudioTrack parseTrack(JsonBrowser json) {
+        if (json.get("available").text().equals("false") || json.get("albums").values().isEmpty()) {
```
Comment (TopiSenpai) [inline main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java:185]: you should use `json.get("available").asBoolean` here

**Review by TopiSenpai — CHANGES_REQUESTED** [review]
else looks good. I assume you tested this as i can't

**Comment by BrokenFG** [comment]
> > Fix error when track doesn't include in any albums (for example user uploaded track manually)
> 
> can we not stream it in this case?

We won't, we check the track for availability and for its presence in the album, if any of these conditions are not met, we skip this track

**Comment by BrokenFG** [comment]
> I assume you tested this as i can't

I came across these errors when trying to play my playlist, after these changes everything works fine for me

File: main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java (hunk 1 above)
Comment (BrokenFG) [reply main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java]: https://github.com/TopiSenpai/LavaSrc/pull/31/commits/a4360ebb824258bd84b42216cbea910a05ac3dd5

**Review by TopiSenpai — APPROVED** [review]
looks good from me. will wait for @AgutinVBoy before merging

File: main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java [hunk 2]
```diff
@@ -173,22 +173,29 @@ public String getDownloadStrings(String uri) throws IOException {
     private List<AudioTrack> parseTracks(JsonBrowser json) {
         var tracks = new ArrayList<AudioTrack>();
         for (var track : json.values()) {
-            tracks.add(this.parseTrack(track));
+            var parsedTrack = this.parseTrack(track);
+            if (parsedTrack != null) {
+                tracks.add(this.parseTrack(track));
```
[the remark this answers is not in the document]
Comment (AgutinVBoy) [reply main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java:178]:
```suggestion
                tracks.add(parsedTrack);
```

**Review by AgutinVBoy — CHANGES_REQUESTED** [review]

**Comment by BrokenFG** [comment]
> Can you provide a case? When I tested the library, I could not get any local track in response.

track must be uploaded to the account whose token is being used, idk what can i provide except of returning json result
https://pastebin.com/c15QWcy3

File: main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (BrokenFG) [reply main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java]: Sorry, I have no idea how I could've missed this

[the remark this answers is not in the document]
Comment (BrokenFG) [reply main/src/main/java/com/github/topisenpai/lavasrc/yandexmusic/YandexMusicSourceManager.java]: Sorry, I have no idea how I could've missed this

**Review by AgutinVBoy — APPROVED** [review]

**Comment by TopiSenpai** [comment]
thx

**Pull request merged**